webhook/resourcesemantics/validation/validation_admit.go

```diff
@@ -141,13 +141,13 @@ func (ac *reconciler) decodeRequestAndPrepareContext(
 		ctx = apis.WithDryRun(ctx)
 	}
 
-	if newObj != nil && oldObj != nil && req.SubResource == "" {
-		ctx = apis.WithinSubResourceUpdate(ctx, oldObj, req.SubResource)
-	}
-
 	switch req.Operation {
 	case admissionv1.Update:
-		ctx = apis.WithinUpdate(ctx, oldObj)
+		if req.SubResource != "" {
+			ctx = apis.WithinSubResourceUpdate(ctx, oldObj, req.SubResource)
```

Comment:
Hmm, does this return the right thing for `apis.IsInUpdate`?

Reply:
yes:
https://github.com/knative/pkg/pull/2546/files/3e662c0cea7c64938e64f45d5c46fef3b4c5741a